### DISCUSSION TITLE
SFDC Library with sales force

### DISCUSSION THREAD

**Discussion by MariamEdward** [pkozuchowski/Apex-Opensource-Library#4] [Q&A]
How can I use the library with my sales force code?

**Comment by pkozuchowski** [comment]
Well you just copy over the components you need to your project repository and deploy :) If you are using old metadata format, then you need may need to convert it before with `sfdx force:source:convert` command.
https://developer.salesforce.com/docs/atlas.en-us.sfdx_cli_reference.meta/sfdx_cli_reference/cli_reference_force_source.htm#cli_reference_force_source_convert

**Reply by MariamEdward** [reply]
Sorry I still didn`t git it .. can you please give me a clear steps to deploy your library to my salesforce sandbox!
 so I can use collection extension methods

**Reply by pkozuchowski** [reply]
Sorry it took me so long to reply. 

I assume you use some kind of IDE - VSCode or IntelliJ to develop in salesforce. 
First you need to clone the repository with `git clone https://github.com/pkozuchowski/Apex-Opensource-Library.git` 
Then you have 2 options:
1) Is simple copy paste the classes you need to your project folder and then deploy with the IDE
2) Use sfdx to deploy collection classes:
` sfdx force:source:deploy -u sandboxName-p "./force-app/commons/core/classes/collections/, ./force-app/commons/core/classes/runtime"`

If you want to deploy entire library, then just do `sfdx force:source:deploy -u scratch -p force-app/`
